### PR TITLE
Tigers - Kat H.

### PR DESCRIPTION
**Pull request by kathawkins**

### CODE REVIEW

File: src/adagrams.js [hunk 1]
```diff
@@ -1,15 +1,101 @@
+
+const letterPool = Array(9).fill("A").concat(Array(2).fill("B")).concat(Array(2).fill("C")).concat(Array(4).fill("D")).concat(Array(12).fill("E")).concat(Array(2).fill("F")).concat(Array(3).fill("G")).concat(Array(2).fill("H")).concat(Array(9).fill("I")).concat(Array(1).fill("J")).concat(Array(1).fill("K")).concat(Array(4).fill("L")).concat(Array(2).fill("M")).concat(Array(6).fill("N")).concat(Array(8).fill("O")).concat(Array(2).fill("P")).concat(Array(1).fill("Q")).concat(Array(6).fill("R")).concat(Array(4).fill("S")).concat(Array(6).fill("T")).concat(Array(4).fill("U")).concat(Array(2).fill("V")).concat(Array(2).fill("W")).concat(Array(1).fill("X")).concat(Array(2).fill("Y")).concat(Array(1).fill("Z"));
+const scoreDict = { 'A': 1,
```
Comment (CheezItMan) [inline src/adagrams.js:3]: Nice solution for scoring the letters.

File: src/adagrams.js [hunk 2]
```diff
@@ -1,15 +1,101 @@
...
+                    'S': 1,
+                    'T': 1, 
+                    'U': 1, 
+                    'V': 4, 
+                    'W': 4, 
+                    'X': 8, 
+                    'Y': 4, 
+                    'Z': 10}
+
 export const drawLetters = () => {
```
Comment (CheezItMan) [inline src/adagrams.js:30]: 👍

File: src/adagrams.js [hunk 3]
```diff
@@ -1,15 +1,101 @@
...
+  for (let i = 0; i < 10; i++){
+    let randLetter = revLetterPool[Math.floor(Math.random() * revLetterPool.length)];
+    let randLetterIndex = revLetterPool.indexOf(randLetter);
+    revLetterPool.splice(randLetterIndex, 1);
+    hand.push(randLetter)
+  }
+  return hand;
 };
 
 export const usesAvailableLetters = (input, lettersInHand) => {
```
Comment (CheezItMan) [inline src/adagrams.js:43]: 👍

File: src/adagrams.js [hunk 4]
```diff
@@ -1,15 +1,101 @@
...
+    if (handCount[letter] === 0) {
+      return false;
+    } else {
+      handCount[letter] -= 1;
+    }
+  }
+  return true;
 };
 
 export const scoreWord = (word) => {
```
Comment (CheezItMan) [inline src/adagrams.js:69]: 👍

File: src/adagrams.js [hunk 5]
```diff
@@ -1,15 +1,101 @@
...
 
 export const highestScoreFrom = (words) => {
   // Implement this method for wave 4
+  let winningScore = 0;
+  let winningWord = "";
+  for (const word of words) {
+    let currentScore = scoreWord(word);
+    if (word.length === 10) {
+      return {word:word, score:currentScore};
+    }
```
Comment (CheezItMan) [inline src/adagrams.js:91]: ⚠️ What if you have 2 10-letter words and a later word has a higher score?

**Review by CheezItMan — COMMENTED** [review]
Nicely done Kat, most of the functions look clean and efficient.  I like the use of a JS dictionary for scoring words.  I did find one issue with `highestScoreFrom` because you missed an edge-case.  That might be a good candidate for a test.
